File: assignment-18/task2.java
public class NumberChecker {

    /**
     * Checks if a given integer is positive, negative, or zero and prints the result.
     * @param num The integer to check.
     */
    public static void checkNumber(int num) {
        if (num > 0) {
            System.out.println("The number is positive");
        } else if (num < 0) {
            System.out.println("The number is negative");
        } else {
            System.out.println("The number is zero");
        }
    }

    public static void main(String[] args) {
        System.out.println("--- Java Execution ---");

        System.out.print("Input: -5 -> Output: ");
        checkNumber(-5);

        System.out.print("Input: 0 -> Output: ");
        checkNumber(0);

        System.out.print("Input: 7 -> Output: ");
        checkNumber(7);
    }
}

File: assignment-18/task2.py
def check_number(num):
    """
    Checks if a given number is positive, negative, or zero and prints the result.
    :param num: The number to check.
    """
    if num > 0:
        print("The number is positive")
    elif num < 0:
        print("The number is negative")
    else:
        print("The number is zero")

# --- Python Execution ---
print("--- Python Execution ---")

print("Input: -5 -> Output: ", end="")
check_number(-5)

print("Input: 0 -> Output: ", end="")
check_number(0)

print("Input: 7 -> Output: ", end="")
check_number(7)
